src/wp-includes/media.php

```diff
@@ -1911,6 +1918,18 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 	return $image;
 }
 
+/**
+ * Adds `decoding=async` attribute to an `img` HTML tag.
+ *
+ * @since 6.0.0
+ *
+ * @param string $image The HTML `img` tag where the attribute should be added.
+ * @return string Converted `img` tag with `decoding=async` attribute added.
+ */
+function wp_img_tag_add_decoding_async_attr( $image ) {
+	return str_replace( '<img ', '<img decoding="async" ', $image );
```

Comment:
Would be great to introduce a filter here to change the default value used for the `decoding` attribute:

```suggestion
	$decoding_value = apply_filters( 'wp_img_tag_add_decoding_async_attr_default', 'async' );
	return str_replace( '<img ', '<img decoding="' . esc_attr(  $decoding_value ) . '" ', $image );
```

Reply:
In addition to the above it would be great to introduce a way to disable this behavior same as happens with the `wp_img_tag_add_width_and_height_attr`.


```suggestion
	$add = apply_filters( 'wp_img_tag_add_decoding_async_attr', true, $image );
	
	if ( ! $add ) {
	  return $image;	
	}
	
	$decoding_value = apply_filters( 'wp_img_tag_add_decoding_async_attr_default', 'async' );
	return str_replace( '<img ', '<img decoding="' . esc_attr(  $decoding_value ) . '" ', $image );
```

Reply:
If that is done, then probably it would be necessary to pass the `$context` in the same way as it is passed to `wp_img_tag_add_loading_attr()`.

Reply:
```suggestion
	/**
	 * Filters the `decoding` attribute value to add to an image. Default `async`.
	 *
	 * Returning `false` or an empty string will not add the attribute.
	 *
	 * @since 6.0.0
	 *
	 * @param string|bool $value   The `decoding` attribute value. Returning a falsey value will result in
	 *                             the attribute being omitted for the image.
	 * @param string      $image   The HTML `img` tag to be filtered.
	 * @param string      $context Additional context about how the function was called or where the img tag is.
	 */
	$value = apply_filters( 'wp_img_tag_decoding_attr', 'async', $image, $context );
	if ( ! $value || ! in_array( $value, array( 'async', 'sync' ), true ) ) {
		return $image;
	}

	return str_replace( '<img ', '<img decoding="' . esc_attr( $value ) . '" ', $image );
```

Reply:
Yea, I agree a filter here would be useful. Some people are sure to want to turn the feature off, and others may have a use for another decoding strategy. Will add Weston's suggestion and review again.